### PR TITLE
add figpager as a third party package 

### PR DESCRIPTION
**Pull request by ebenp**
## PR Summary
I wrote a class around matplotlib's single and multi page backends to control subplot layout and some figure elements such s text, lines and images to make single and multi page figure templates. This PR adds links to this template library to the third party documentation page in matplotlib's documentation.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [x] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/next_api_changes/* if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by story645** [comment]
please add `Framework :: Matplotlib` to the classifiers list in setup.py

**Comment by ebenp** [comment]
Thanks! :) I've updated setup.py 
[https://github.com/ebenp/figpager/commit/245ab1d743d83801250b06a7367e4a151b5dc584](https://github.com/ebenp/figpager/commit/245ab1d743d83801250b06a7367e4a151b5dc584)

File: doc/thirdpartypackages/index.rst [hunk 1]
```diff
@@ -282,6 +282,14 @@ border, and logo.
 .. image:: /_static/mpl_template_example.png
     :height: 330px
 
+figpager
+=====
```
Comment (jklymak) [inline doc/thirdpartypackages/index.rst:286]: undescores must span the element above.

File: doc/thirdpartypackages/index.rst [hunk 2]
```diff
@@ -282,6 +282,14 @@ border, and logo.
...
+
+`figpager <https://pypi.org/project/figpager/>`_ provides customizable figure elements such as text, lines and images
```
Comment (jklymak) [inline doc/thirdpartypackages/index.rst:288]: Please wrap at 79 char....

File: doc/thirdpartypackages/index.rst (hunk 1 above)
Comment (ebenp) [reply doc/thirdpartypackages/index.rst]: Thanks! I've extended them below.
[https://github.com/ebenp/matplotlib/commit/898e67f6517a0744a7c39596b757d15068b44f3d](https://github.com/ebenp/matplotlib/commit/898e67f6517a0744a7c39596b757d15068b44f3d)

File: doc/thirdpartypackages/index.rst (hunk 2 above)
Comment (ebenp) [reply doc/thirdpartypackages/index.rst]: Thanks! I've wrapped it below.
[https://github.com/ebenp/matplotlib/commit/898e67f6517a0744a7c39596b757d15068b44f3d](https://github.com/ebenp/matplotlib/commit/898e67f6517a0744a7c39596b757d15068b44f3d)

**Comment by ebenp** [comment]
> We try to keep image sizes small. Please run the image though a png compressor like https://tinypng.com/.

Thanks for thinking of the pixels! Thanks for linking to a compressor as well. I think I've updated things correctly.


Smaller image processed through tinypng and uploaded.
[https://github.com/ebenp/matplotlib/commit/f8d7f9a31b4d4877e217545ca14b0bb0b9a758f4](https://github.com/ebenp/matplotlib/commit/f8d7f9a31b4d4877e217545ca14b0bb0b9a758f4)

Image path update in index.rst
[https://github.com/ebenp/matplotlib/commit/37a6fffb278396c222953123531cff2b4a9b401d#diff-688463b7d6617526f45f8caa3c0158c6](https://github.com/ebenp/matplotlib/commit/37a6fffb278396c222953123531cff2b4a9b401d#diff-688463b7d6617526f45f8caa3c0158c6)

**Comment by tacaswell** [comment]
@ebenp you may also want to put in a JOSS (https://joss.theoj.org/) paper?

File: doc/thirdpartypackages/index.rst [hunk 3]
```diff
@@ -283,10 +283,11 @@ border, and logo.
...
 
 figpager
-=====
+========
 
-`figpager <https://pypi.org/project/figpager/>`_ provides customizable figure elements such as text, lines and images
-and subplot layout control for single or multi page output
+`figpager <https://pypi.org/project/figpager/>`_ provides 
+customizable figure elements such as text, lines and images
+and subplot layout control for single or multi page output.
```
Comment (story645) [inline doc/thirdpartypackages/index.rst:290]: `and a subplot layout`

**Comment by jklymak** [comment]
All these commits should be squashed before merging....

**Comment by ebenp** [comment]
Ugh. Tried to squash / rebase and utterly failed. I kept getting newer matplotlib changes interspersed. Anyway, I nuked and paved by re forking and updating the files. How do I reset the link to the respoitory? HALP. :)

[https://github.com/ebenp/matplotlib/commit/1b783f1e13ea7cea8bb387fdb1430ee2db3b4e59](https://github.com/ebenp/matplotlib/commit/1b783f1e13ea7cea8bb387fdb1430ee2db3b4e59)

**Comment by jklymak** [comment]
So sorry! I should have been more clear that *you* didn't need to do the squash if you weren't comfortable with the process.

**Comment by ebenp** [comment]
> So sorry! I should have been more clear that _you_ didn't need to do the squash if you weren't comfortable with the process.

Oh, no worries. I thought I could figure it out. I think I was close but my changes kept editing other files; _init_.py and font_manager and I got worried. For the future I'll ask from help in squashing. I assume I pull and then squash? Or do I squash my commits and don't worry about it being out of sync with the current matplotlib?

**Comment by jklymak** [comment]
I typically `git rebase upstream/master`, fix any conflicts, and then `git log` to see what the last commit on master is (say xxxx) and then do `git rebase -i xxxx`.  https://matplotlib.org/3.3.0/devel/gitwash/index.html should go through this. If its not clear, let us know...

**Comment by ebenp** [comment]
> I typically `git rebase upstream/master`, fix any conflicts, and then `git log` to see what the last commit on master is (say xxxx) and then do `git rebase -i xxxx`. https://matplotlib.org/3.3.0/devel/gitwash/index.html should go through this. If its not clear, let us know...

Thanks! That is super helpful. For this draft pull request is everything ok despite me removing the repository? Should I recreate the draft pull request or create a PR?

**Comment by ebenp** [comment]
> @ebenp you may also want to put in a JOSS (https://joss.theoj.org/) paper?

Great idea! I'll have to come up with a spiffy logo!

**Comment by timhoffm** [comment]
Generally, you don't need to worry about syncing. The GitHub UI will tell you if there is a merge conflict. And only then you need to sync (by rebasing on top of master).

**Comment by ebenp** [comment]
> Generally, you don't need to worry about syncing. The GitHub UI will tell you if there is a merge conflict. And only then you need to sync (by rebasing on top of master).

Ah ok. Yeah, I tried to be all fancy before I squashed through rebase and quickly was out of my depth. I won't worry about it in the future.

Thanks for the help and support all! This has been a really welcoming intro to matplotlib PRs.

**Comment by ebenp** [comment]
Let me know if there's a way to relink my repository or if I should create a new PR.

**Comment by QuLogic** [comment]
I don't think there is a way. Probably be best to create a new PR if this one is no longer updating correctly.

**Comment by ebenp** [comment]
Thanks! Sounds great. A new PR is here: [https://github.com/matplotlib/matplotlib/pull/18059](https://github.com/matplotlib/matplotlib/pull/18059)

**Pull request closed without merging**